### ISSUE TITLE
Error for the code with upsampling2D

### ISSUE THREAD

**Issue by spiking-neuron** [Dobiasd/frugally-deep#182]
Hi, thanks for very nice library.

I read the discussion in #157 but I still have error which is probably due to upsampling2D.

My keras and tensorflow versions are Keras version: 2.2.4 TensorFlow version: 1.13.1.
I use Python 3.6.8 in WSL Ubuntu.
I think I installed frugally-deep successfully and the example code in the main page (dense NN is used) gives no error. 
However, my CNN code gives errors and I recognized  that  upsampling2D was the cause (no error if only conv2D and maxpooling2D are used). 
I tried the simple code with upsampling2D in #157 but it also gave the following error (I put #define NDEBUG line at the top of C++ code and compiled by  g++ -std=c++14 main.cpp   ): 

Loading json ... done. elapsed time: 0.002509 s
terminate called after throwing an instance of 'nlohmann::detail::type_error'
  what():  [json.exception.type_error.302] type must be string, but is number
Aborted (core dumped)

I'm struggling with this problem for several days but I still cannot resolve it. 
model.json file created by convert_model.py is attached. 
Any help would be appreciated.

[model.json.txt](https://github.com/Dobiasd/frugally-deep/files/3662400/model.json.txt)

**Comment by Dobiasd** [comment]
Hi,

and thanks for the good report.

I can reproduce the exception in C++ with your `model.json` file.

```
ERROR: [json.exception.type_error.302] type must be string, but is number
```

It originates from the following line in `create_upsampling_2d_layer`:

```cpp
const std::string interpolation = data["config"]["interpolation"];
```

Looking at the respective object in the JSON file, `interpolation` is missing indeed:

```json
        {
          "class_name": "UpSampling2D",
          "config": {
            "data_format": "channels_last",
            "dtype": "float32",
            "name": "up_sampling2d",
            "size": [
              2,
              2
            ],
            "trainable": true
          },
          "inbound_nodes": [
            [
              [
                "input_1",
                0,
                0,
                {}
              ]
            ]
          ],
          "name": "up_sampling2d"
        }
```

The default value for the `interpolation` argument in Keras's `UpSampling2D` constructor is `'nearest'`.

So in general, this should work, and there should always be an interpolation type specified. I just verified by just converting such a model from `h5` to `json`.

Could you please try to update your versions of Keras and TensorFlow to `1.14.0` and `2.3.0` and then try again with your model, i.e., re-converting it with `convert_model.py`, etc.?

Also, make sure, you are using the latest version of frugally-deep.

If this all does not help, please upload your `h5` file too, or a minimal Python-code example to re-create the problematic model.

**Comment by spiking-neuron** [comment]
Hi, Thank you for reply and for detailed explanation!

I first inserted
 "interpolation": "nearest", 
in the UpSampling2D part of my model.json by hand and succeeded to run the program.

As for the version issue, in my previous try I have actually replaced the following lines: 
import keras
from keras import backend as K
from keras.layers import Input
from keras.models import Model, load_model

in the file  convert_model.py by the following:

from tensorflow import keras
from tensorflow.keras import backend as K
from tensorflow.keras.layers import Input
from tensorflow.keras.models import Model, load_model
without installing keras itself. and I experienced the error which I explained. 

This time I installed keras 2.3.0 by pip install keras, and I used the original convert_model.py to generate model.json. Then the generated model.json does certainly include "interpolation": "nearest", , and now I succeeded to run the c++ program. 
Now everything is fine! Thank you so much again.
